File: examples/javascript/get_solutions.js
const BASE_URL = "https://wattcoin-production-81a7.up.railway.app";

async function getSolutions() {
  try {
    const response = await fetch(`${BASE_URL}/solutions`);
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }
    const data = await response.json();
    console.log(data);
  } catch (error) {
    console.error("Error fetching solutions:", error);
  }
}

getSolutions();

File: examples/python/get_solutions.py
import requests

BASE_URL = "https://wattcoin-production-81a7.up.railway.app"

def get_solutions():
    """Fetch submitted solutions."""
    response = requests.get(f"{BASE_URL}/solutions")
    response.raise_for_status()
    return response.json()

if __name__ == "__main__":
    solutions = get_solutions()
    print(solutions)
